clmm/support/sampler.py

```diff
@@ -4,32 +4,129 @@
+        Please check the `scipy` documentaion for information on default values and methods.
+        
+    Returns
+    -------
+    `x`` : array
+    The solution of the optimization    
+    
+    '''
+    
     from scipy import optimize as spo
 
-    return spo.basinhopping(model_to_shear_profile, logm_0, minimizer_kwargs={'args':args}).x[0]
+    return spo.basinhopping(model_to_shear_profile, logm_0, **kwargs).x
 
-def scicurve_fit(profile_model,radius,profile,err_profile,bounds=None,p0=None):
-    '''Uses scipy.optimize.curve_fit to find best fit parameters'''
+
+def scicurve_fit(profile_model,radius,profile,err_profile, absolute_sigma = True, **kwargs):
+    r"""
+    Uses scipy.optimize.curve_fit to find best fit parameters
+    
+    Parameters
+    ----------
+    profile_model : callable
+        The model function, f(x, ...). It must take the independent
+        variable as the first argument and the parameters to fit as
+        separate remaining arguments.
+    radius : array_like or object
+        The independent variable where the data is measured.
+        Should usually be an M-length sequence or an (k,M)-shaped array for
+        functions with k predictors, but can actually be any object.
+    profile : array_like
+        The dependent data, a length M array - nominally ``f(xdata, ...)``.
+    err_profile : M-length sequence or MxM array
+        Determines the uncertainty in `ydata`. If we define residuals as
+        ``r = ydata - f(xdata, *popt)``, then the interpretation of `sigma`
+        depends on its number of dimensions:
+        - A 1-D `sigma` should contain values of standard deviations of
+        errors in `ydata`. In this case, the optimized function is
+        ``chisq = sum((r / sigma) ** 2)``.
+        - A 2-D `sigma` should contain the covariance matrix of
+        errors in `ydata`. In this case, the optimized function is
+        ``chisq = r.T @ inv(sigma) @ r``.
+    absolute_sigma : bool, optional
+        If True (default), `sigma` is used in an absolute sense and the estimated parameter
+        covariance `pcov` reflects these absolute values.
+        If False, only the relative magnitudes of the `sigma` values matter.
+        The returned parameter covariance matrix `pcov` is based on scaling
+        `sigma` by a constant factor. This constant is set by demanding that the
+        reduced `chisq` for the optimal parameters `popt` when using the
+        *scaled* `sigma` equals unity. In other words, `sigma` is scaled to
+        match the sample variance of the residuals after the fit. Default is True.
+        Mathematically,
+        ``pcov(absolute_sigma=False) = pcov(absolute_sigma=True) * chisq(popt)/(M-N)``
+        
+    kwargs :
+        Other optional keyword arguments passed to `curve_fit` 
+        Please check the `scipy` documentaion for information on default values and methods.
+        
+    Returns
+    -------
+    p : list
+        contains : 
```

Comment:
I am thinking if it would be better to still call the two arrays popt and pcov, since these two names occur in the context.

Reply:
Yes, I agree but I kept it like that  in order not to change all the notebooks.

Reply:
@hsinfan1996 I think this wrapper around a fitter will ultimately be removed even from `support`, but this quick fix is simpler for now.

Reply:
I see.